### PR TITLE
updated the constructor call for schema in models in accordance with …

### PR DESCRIPTION
**Pull request by VaibhavAjayGupta**
Constructor call for creating new schema in models is not in accordance with javascript. So updated them by adding new operator.

### CODE REVIEW

**Comment by hamishwillee** [comment]
Thank you. This is not required - the function acts as a constructor to create schema whether or not you put new in front of it. That said it is probably clearer and should not do any harm. I'll check it over the coming month.

**Comment by VaibhavAjayGupta** [comment]
Thank you for your feedback.

**Comment by VaibhavAjayGupta** [comment]
Fixed some depreciated errors and warnings.
1. .isDate() is removed from validator.js in version 7.(error)
2. mongoose.connect requires useMongoClient: true. (warning)
3. mpromise depreciated so added the default promise. (warning)
4. updated the .gitignore file to ignore package-lock.json file.

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,9 @@ logs
 *.log
 npm-debug.log*
 
+# Lock files
+package-lock.json
```
Comment (hamishwillee) [inline .gitignore]: I think it makes sense for us to include these, to allow users to "guarantee" they are using the right dependencies.

**Comment by hamishwillee** [comment]
@VaibhavAjayGupta This is all very helpful. Unfortunately you've mixed changes across different parts of the system in one PR. That makes this very hard for me to merge, because I can't easily validate and update the MDN documentation at the same time. For future reference can you please split PRs by "functionality" - so mongo db changes would not get mixed with validation changes.

I will probably have to manually copy the changes in in order to manage the review process. That sux, because you wouldn't get "credit"

**Comment by hamishwillee** [comment]
OK, I am cherry picking your fixes thanks. You can watch them go in as https://github.com/mdn/express-locallibrary-tutorial/pull/16 . Phew, wanted to make sure you do get recognised as the contributor :-)

**Comment by hamishwillee** [comment]
... and now I see you've updated the documentation as well. Thank you so much!

File: .gitignore (hunk 1 above)
Comment (VaibhavAjayGupta) [reply .gitignore]: I thought the same but wasn't certain that's why added to the ignore list.

**Comment by VaibhavAjayGupta** [comment]
Thank you for your effort, time and feedback. Sorry for the mixing. Will take care in future :)

**Comment by hamishwillee** [comment]
No way - thank YOU for doing this work. It has been very helpful for the review. I'll integrate the next pr when I get up to the validation topic.

**Pull request closed without merging**